### PR TITLE
Search: fix highlighting negated predicates

### PR DESCRIPTION
**Pull request by camdencheek**
Syntax highlighting of predicates currently fails. This fixes highlighting negated predicates. It makes no effort to disambiguate between which predicates support highlighting and which don't.

## Test plan

Manual testing.

Before:
Screen Shot 2022-09-09 at 16 20 49

After:
Screen Shot 2022-09-09 at 16 21 17

## App preview:

- [Web](https://sg-web-cc-negated-predicate-highlighting.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-sftgidlpjx.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/shared/src/search/query/decoratedToken.ts [hunk 1]
```diff
@@ -951,7 +951,6 @@ const decorateRepoHasBody = (body: string, offset: number): DecoratedToken[] | u
     if (!matches) {
         return undefined
     }
-    console.log(matches)
```
Comment (camdencheek) [inline client/shared/src/search/query/decoratedToken.ts:954]: This is an unrelated oops from a previous PR 🙈

File: client/shared/src/search/query/predicates.ts [hunk 2]
```diff
@@ -149,6 +149,10 @@ export const scanPredicate = (field: string, value: string): Predicate | undefin
     }
     const name = match[0]
     const path = name.split('.')
+    // Remove negation from the field for lookup
+    if (field[0] == '-') {
+        field = field.substring(1)
+    }
```
Comment (camdencheek) [inline client/shared/src/search/query/predicates.ts:155]: The issue is that `field` will contain the `-` like `-repo`. We use this string to do a lookup of supported predicates, and the `-` breaks the comparison. 

Also, is this the best way to remove a prefix from a javascript string?

Comment (tbliu98) [reply client/shared/src/search/query/predicates.ts:155]: There is a `slice()` method for strings but it seems to be virtually identical to `substring()` https://stackoverflow.com/questions/2243824/what-is-the-difference-between-string-slice-and-string-substring

**Review by tbliu98 — APPROVED** [review]

File: client/shared/src/search/query/predicates.ts (hunk 2 above)
Comment (camdencheek) [reply client/shared/src/search/query/predicates.ts]: I think I've been spoiled by `.strip_prefix()` methods in other languages 😄
